### PR TITLE
Use records in query news

### PR DESCRIPTION
**Pull request by fax4ever**

### CODE REVIEW

File: _posts/2024-06-10-infinispan-15-queries.adoc [hunk 1]
```diff
@@ -28,16 +28,7 @@ Following an example of entity, query and result:
...
-   private final String description;
-
-   @ProtoFactory
-   public Game(String name, String description) {
-      this.name = name;
-      this.description = description;
-   }
+public record Game(String name, String description) {
 
    @Keyword(projectable = true, sortable = true)
```
Comment (tristantarrant) [inline _posts/2024-06-10-infinispan-15-queries.adoc:33]: You want to get rid of the getters too, and put the annotations on the record field declaration

Comment (fax4ever) [reply _posts/2024-06-10-infinispan-15-queries.adoc:33]: Need to test it first. Then of course I'll do it

Comment (tristantarrant) [reply _posts/2024-06-10-infinispan-15-queries.adoc:33]: Example:
```java
@Proto
@Indexed
public record Book(
    @Text String title,
    @Keyword(projectable = true, sortable = true, normalizer = "lowercase", indexNullAs = "unnamed", norms = false) String description,
    int publicationYear,
    Set<Author> authors,
    Type bookType,
    BigDecimal price
) { }
```

Comment (fax4ever) [reply _posts/2024-06-10-infinispan-15-queries.adoc]: Tested. It works nicely! Pushed the changes

**Comment by fax4ever** [comment]
Thanks for the suggestion @tristantarrant.
With these changes the API improved a lot.

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**